cobalt/browser/global_features.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COBALT_BROWSER_GLOBAL_FEATURES_H_
+#define COBALT_BROWSER_GLOBAL_FEATURES_H_
+
+#include "base/memory/raw_ptr.h"
+#include "base/sequence_checker.h"
+
+class PrefService;
+
+namespace metrics {
+class MetricsService;
+}  // namespace metrics
+
+namespace metrics_services_manager {
+class MetricsServicesManager;
+}  // namespace metrics_services_manager
+
+namespace cobalt {
+class CobaltMetricsServicesManagerClient;
+
+// This class owns features that are globally scoped. It follows the structure
+// of BrowserProcess class in Chrome.
+class GlobalFeatures {
+ public:
+  GlobalFeatures(
+      std::unique_ptr<PrefService> experiment_config,
+      std::unique_ptr<PrefService> local_state,
+      std::unique_ptr<metrics_services_manager::MetricsServicesManager> manager,
+      CobaltMetricsServicesManagerClient* client);
+
+  ~GlobalFeatures();
+
+  metrics_services_manager::MetricsServicesManager* GetMetricsServicesManager();
```

Comment:
Can we add unittest for this class please? It can be very simple, e.g. one test case that constructs-destroys this class (we can check that getters return the expected values), another where we pass a null `MetricsServicesManager` on construction and then call `GetMetricsServicesManager()` twice (the first time it returns a non-null pointer, and the second time returns the same pointer as before). Just some ideas. 

Note that you don't need to create a test-class for something so simple, it can be just TEST() cases and then generate the necessary objects on the spot.

Reply:
Yes, absolutely! FWIW, Yijia and I planned testing as a whole project/phase on its own shortly after submitting the core stuff under https://b.corp.google.com/issues/407734389. That way I can be unblocked on the code that needs this in UMA H5vcc.

Reply:
Gotcha. Please leave a `// TODO(b/392683875): Add unittests` in the class declaration so it's not forgotten.